features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/ServiceCatalog/Listing/Delete.jsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import { FormattedMessage, injectIntl, useIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import Icon from '@material-ui/core/Icon';
+import Alert from 'AppComponents/Shared/Alert';
+import ConfirmDialog from 'AppComponents/Shared/ConfirmDialog';
+import ServiceCatalog from 'AppData/ServiceCatalog';
+
+/**
+* Service Catalog service delete
+* @param {any} props Props for delete function.
+* @returns {any} Returns the rendered UI for service delete.
+*/
+function Delete(props) {
+    const intl = useIntl();
+    const [open, setOpen] = useState(false);
+    const toggleOpen = () => {
+        setOpen(!open);
+    };
+    const { serviceName, serviceId, getData } = props;
```

Comment:
This set of properties (or in short `props`) are the parameters that external component can engage with this Delete component, In other words, this act as the interface for this `Delete` component(if we look at it in OOP corner)
Ie : if `Delete` is a class `serviceName, serviceId, getData` are like constructor arguments.

```java
Delete(serviceName, serviceId, getData)
```
So my point here is, For this Delete component `getData` kind of function property is not appropriate.
Usually, the component states are event-driven like, onChange, onClose, onSubmit, etc so it's better to have an event handler like onDelete and implement the onDelete handler in parent component.

Delete component should ask for user content and if the user confirms the action it should trigger the `onDelete` handler

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9407/commits/56c6c22eb83606fd6816a62432d941fba1aceafb